core/utils/big_math/big_math.go

```diff
@@ -51,6 +51,17 @@ func Max(x, y interface{}) *big.Int {
 	return yBig
 }
 
+// Min returns the min of the two given values after coercing them to big.Int,
+// or panics if it cannot.
+func Min(x, y interface{}) *big.Int {
```

Comment:
I would make this explicit and not do any coercion. `*big.Int` everywhere.

Reply:
I will leave as is for now to be consistent with existing methods in this file. e.g. [Max](https://github.com/smartcontractkit/chainlink/blob/develop/core/utils/big_math/big_math.go#L45) function does coercion.

Reply:
Can it be done with generics instead?

Reply:
It doesn't seem like there is a straight-forward way to do it using generics. If the numeric types that we used are internal, then we can do something like:

```
func Min[Number any](x, y Number) *big.Int {
   return Min(x, y) 
}
```
But in this case, the types are either primitive or `big`. 

If we really want to avoid coercion, I think using *big.Int will be the best. Will leave this refactoring out of scope of this change.